src/main/scala/org/apache/sysml/api/ml/BaseSystemMLClassifier.scala

```diff
@@ -217,25 +217,34 @@ trait BaseSystemMLClassifier extends BaseSystemMLEstimator {
 
 trait BaseSystemMLClassifierModel extends BaseSystemMLEstimatorModel {
 
-  def baseTransform(X: MatrixBlock, sc: SparkContext, probVar:String): MatrixBlock = {
+	def baseTransform(X: MatrixBlock, sc: SparkContext, probVar:String): MatrixBlock = baseTransform(X, sc, probVar, -1, 1, 1)
+	
+	def baseTransform(X: MatrixBlock, sc: SparkContext, probVar:String, C:Int, H: Int, W:Int): MatrixBlock = {
     val isSingleNode = true
     val ml = new MLContext(sc)
     updateML(ml)
     val script = getPredictionScript(isSingleNode)
     // Uncomment for debugging
     // ml.setExplainLevel(ExplainLevel.RECOMPILE_RUNTIME)
     val modelPredict = ml.execute(script._1.in(script._2, X, new MatrixMetadata(X.getNumRows, X.getNumColumns, X.getNonZeros)))
-    val ret = PredictionUtils.computePredictedClassLabelsFromProbability(modelPredict, isSingleNode, sc, probVar)
-              .getMatrix("Prediction").toMatrixBlock
+    val Prob = modelPredict.getMatrix(probVar)
+    val script1 = dml("source(\"nn/util.dml\") as util; Prediction = util::predict_class(Prob, C, H, W);")
```

Comment:
If it is helpful here, @feihugis added [`top_k`](https://github.com/apache/systemml/blob/master/scripts/nn/util.dml#L303) and [`top_k2d`](https://github.com/apache/systemml/blob/master/scripts/nn/util.dml#L330) functions that can compute the top k desired predictions, returning the class and associated probability values.